modules/shared/src/main/scala/org/tessellation/currency/dataApplication/package.scala

```diff
@@ -130,10 +132,12 @@ trait DataApplicationContextualOps[F[_], D <: DataUpdate, DON <: DataOnChainStat
   def routesPrefix: ExternalUrlPrefix = "/data-application"
 }
 
-trait DataApplicationL0Service[F[_], D <: DataUpdate, DON <: DataOnChainState, DOF <: DataCalculatedState]
+abstract class DataApplicationL0Service[F[_]: Applicative, D <: DataUpdate, DON <: DataOnChainState, DOF <: DataCalculatedState]
```

Comment:
It should stay a trait. Please see me previous comment

Reply:
@marcinwadon If it's a trait then we can't have the `Applicative[F].unit` implementation here, which is why I asked where you want the implementation to go.

Reply:
Yeah we looked over this, if we have another type argument in the function, then we need to translate from G (function arg) to F anyways (hence the problem with the trait). I'm not seeing a way around this otherwise? I do agree with a default from a user perspective being easier (see the example route having a specific impl in the trait,) but the problem here is with the type bounds.

I'm fine with abstract class here if we can't find a cleaner way around this.

Reply:
If you can supply another alternative marcin happy to use this here (which translates from the type bounds on the function type param to the trait param.)

Reply:
@ryle-goehausen I've already written it twice somewhere in comments. In general, it's as simple as:
```
trait DataApplicationL0Service[F[_], D <: DataUpdate, ...] {

  def onConsumeArtifact(snapshot: Signed[CurrencyIncrementalSnapshot])(implicit A: Applicative[F]): F[Unit] = A.unit

}
```

Reply:
@sdavidp /\